static/src/javascripts/lib/string-utils.spec.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+import { optionalStringToNumber } from 'lib/string-utils';
+
+describe('optionalStringToNumber', () => {
+    it('should return a number', () => {
+        expect(optionalStringToNumber('3')).toBe(3);
+    });
+
+    it('should return undefined if undefined', () => {
+        expect(optionalStringToNumber(undefined)).toBe(undefined);
+    });
+
+    it('should return undefined if NaN', () => {
```

Comment:
maybe also add a test for '0' should be 0 ?

Reply:
Also if an integer number is needed you could do `parseInt(str, 10)` rather than Number(str) and then do the NaN check...